podman-desktop: 1.13.2 -> 1.15.0

kachick opened the pull request:
Diff: https://github.com/podman-desktop/podman-desktop/compare/v1.13.2...v1.15.0

Update postPatch with https://github.com/podman-desktop/podman-desktop/commit/8af3761b41f051933726b187617d14227ab0f0b8

This PR aims to fix [failure in nixpkgs-update](https://nixpkgs-update-logs.nix-community.org/podman-desktop/2024-11-26.log)

```
patching file extensions/podman/packages/extension/package.json
substituteStream() in derivation podman-desktop-1.14.1: ERROR: pattern process.resourcesPath doesn't match anything in file 'packages/main/src/tray-animate-icon.ts'
```


<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

kachick committed:
podman-desktop: 1.13.2 -> 1.14.1

Diff: https://github.com/podman-desktop/podman-desktop/compare/v1.13.2...v1.14.1

Update postPatch with
https://github.com/podman-desktop/podman-desktop/commit/8af3761b41f051933726b187617d14227ab0f0b8

This fixes nixpkgs-update failure
https://nixpkgs-update-logs.nix-community.org/podman-desktop/2024-11-26.log

```
patching file extensions/podman/packages/extension/package.json
substituteStream() in derivation podman-desktop-1.14.1: ERROR: pattern process.resourcesPath doesn't match anything in file 'packages/main/src/tray-animate-icon.ts'
```

Pin pnpm and electron version to respect
https://github.com/NixOS/nixpkgs/blob/1c26355e02ea8aa9bef6a7b3f59d74bd3c504c11/doc/languages-frameworks/javascript.section.md?plain=1#L427

Checked the version in

https://github.com/podman-desktop/podman-desktop/blob/v1.14.1/package.json
https://github.com/podman-desktop/podman-desktop/blob/v1.14.1/pnpm-lock.yaml

Attempt to fix the Linux build with inherit finalAttrs.patches

```
error: hash mismatch in fixed-output derivation '/nix/store/xq1bg197mw0a0gr1k06xfr405d03fsg3-podman-desktop-pnpm-deps.drv':
         specified: sha256-f7LDfKMPaVRtgvZLFz2JePXd45JBdBCNQR/Ch6VaBKU=
            got:    sha256-bbfyEpDj85HGC3fNLY2to2nKdldbefklJjWGpAYdSdw=
```

See https://github.com/NixOS/nixpkgs/blob/1c26355e02ea8aa9bef6a7b3f59d74bd3c504c11/doc/languages-frameworks/javascript.section.md?plain=1#L429

However it still making hash mismatch in nixpkgs-review

booxter commented:
Thank you for the PR; thanks for including helpful details about particular changes in the commit message.

The code change look good, and I was able to build both for linux and darwin. The darwin version starts and seems to work fine. My only concern is that the commit message lines are not as wrapped as it could be (I don't see a lot of commits in the repo that have such long lines). But I'm new to nixpkgs maintenance, maybe it's acceptable here. Let me know what you think. Other than that, this is good to go.

kachick committed:
podman-desktop: 1.14.1 -> 1.15.0

Diff: https://github.com/podman-desktop/podman-desktop/compare/v1.14.1...v1.15.0

kachick commented on pkgs/applications/virtualization/podman-desktop/default.nix:
```diff
@@ -3,30 +3,30 @@
-    hash = "sha256-LPsNRd1c/cQeyBn3LZKnKeAsZ981sOkLYTnXIZL82LA=";
+  pnpmDeps = pnpm_9.fetchDeps {
+    inherit (finalAttrs) pname version src patches;
+    hash = "sha256-EzkAcd84JEBq6eDkh9XCGfCyeo5r2vpZSIvb2hqpsBU=";
```
This hash always will be different in nixpkgs-review in x86_64-linux report.

In nixpkgs-review

```
error: hash mismatch in fixed-output derivation '/nix/store/hm84x737h5dnd3g2ky8k5hg6fkmd8ly6-podman-desktop-pnpm-deps.drv':
         specified: sha256-EzkAcd84JEBq6eDkh9XCGfCyeo5r2vpZSIvb2hqpsBU=
            got:    sha256-2wmMuymv8C1ZVeGnWf2R3t9cf1jqiFIw0KJ/m3/cxZs=
error: 1 dependencies of derivation '/nix/store/wgrqfbf8wqkhvk7br091vw2nkfpn94dg-podman-desktop-1.15.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/fiw3n0zkxpwv2zcsc6zc0rya43rs315l-review-shell.drv' failed to build
```

It succeeds in darwin. I have no idea what is this :thinking:

kachick replied:
Might be related to https://github.com/NixOS/nixpkgs/pull/350063?

kachick commented:
@booxter Thank you for your reviewing. :pray: 

I don't have confidence that wrapping long URLs and logs will be better or not in the commit message. So keeping now. But I can modify commit messages if required. Any suggestions welcome.

However running nixpkgs-review always fails for the x86_64-linux and `pnpm_9.fetchDeps` step with hash mismatched. I don't know how to fix this behaviors. :bow:

kachick committed:
podman-desktop: attempt to fix nixpkgs-review failure in x86_64-linux

I can not reproduce in my device, however github hosted ubuntu-24.04 runner always returns different hash for fetchDeps

kachick committed:
podman-desktop: skip to install playwright

kachick commented:
I have no idea how to fix the hash mismatch, at least this year. Sorry for closing this PR 🙇.
